addons/account/models/account_move.py

```diff
@@ -1484,17 +1509,10 @@ def _compute_amount(self):
                     new_pmt_state = 'partial'
 
             if new_pmt_state == 'paid' and move.move_type in ('in_invoice', 'out_invoice', 'entry'):
-                reverse_type = move.move_type == 'in_invoice' and 'in_refund' or move.move_type == 'out_invoice' and 'out_refund' or 'entry'
-                reverse_moves = self.env['account.move'].search([('reversed_entry_id', '=', move.id), ('state', '=', 'posted'), ('move_type', '=', reverse_type)])
```

Comment:
I'm not actually sure why we need that check on the move type. Isn't a reverse entry just a reverse entry?
If we don't need to do this, we can simplify the code above quite a lot.

CC: @oco-odoo since you wrote the original code

Reply:
Need to have the confirmation of oco but I think it's to take advantage of the index on `move_type` making the search faster

Reply:
Doesn't make sense, there is already an index on `reversed_entry_id`, adding multiple clauses like that could almost only make it worse by making the query planner confused about which indexes to use

Reply:
Correct me if I'm wrong but I think it's only the case since 15.2
Here is the field definition for 14.0 https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_move.py#L263-L264
Maybe what we really need is to backport the index in v14 and v15 ?

Reply:
Oh okay then, lets wait for confirmation from OCO and if that's the case, let's leave it until 15.0 an remove after